rust/lance/src/dataset/optimize.rs

```diff
@@ -2069,4 +2075,107 @@ mod tests {
             );
         }
     }
+
+    #[tokio::test]
+    async fn test_remap_index_after_compaction() {
+        let mut data_gen = BatchGenerator::new()
+            .col(Box::new(
+                RandomVector::new().vec_width(128).named("vec".to_owned()),
+            ))
+            .col(Box::new(IncrementingInt32::new().named("i".to_owned())));
+
+        let mut dataset = Dataset::write(
+            data_gen.batch(6_000),
+            "memory://test/table",
+            Some(WriteParams {
+                max_rows_per_file: 1_000, // 6 files
+                ..Default::default()
+            }),
+        )
+        .await
+        .unwrap();
```

Comment:
This is a slightly older way of generating data.  It's still used in a few places and probably ok to keep.  However, the `lance_datagen` utilities are slightly more compact and more capable...

```
            let mut dataset = lance_datagen::gen()
                .col("vec", array::rand_vec(Dimension::from(128))
                .col("i", array::step::<Int32Type>())
                .into_ram_dataset(FragmentCount::from(6), FragmentRowCount::from(1000))
                .await
                .unwrap();
```

Reply:
https://github.com/lancedb/lance/issues/3914